rateLimit.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * ./rateLimit.js
+ *
+ * Rate limiting setup
+ */
+const conf = require('./config');
+const limiterRedisClient = require('./cache/redisCache').client.limiter;
+const Limiter = require('ratelimiter');
+const Promise = require('bluebird');
+Limiter.prototype.getAsync = Promise.promisify(Limiter.prototype.get);
+
+module.exports = function (req, res, next) {
+  if (req.headers.IsAdmin) return next();
+  const limiterConfig1 = {
+    db: limiterRedisClient,
+    max: conf.expressLimiterTotal,
+    duration: conf.expressLimiterExpire,
+  };
+  const limiterConfig2 = {
+    db: limiterRedisClient,
+    max: conf.expressLimiterTotal2,
+    duration: conf.expressLimiterExpire2,
+  };
+
+  const lookupVals = [];
+  const lookups = conf.expressLimiterLookup;
+  lookups.forEach((lookupPath) => {
+    const lookupArr = lookupPath.split('.');
+    const lookupVal = lookupArr.reduce((prev, cur) => prev ? prev[cur] : null, req);
+    if (lookupVal) lookupVals.push(lookupVal);
+  });
+
+  const id = lookupVals.join('_');
+  if (id) {
+    limiterConfig1.id = `1__${id}`;
+    limiterConfig2.id = `2__${id}`;
+  }
+
+  // The order matters. The second one overwrites the limit headers.
+  Promise.resolve()
+  .then(() => applyLimits(limiterConfig2))
```

Comment:
What happens when there is no config for #2?

Reply:
It will be skipped.
Line 56: return Promise.resolve();